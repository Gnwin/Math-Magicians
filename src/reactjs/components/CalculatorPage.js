import React, { useState } from 'react';

import numbers from './numbers';
import operators from './operators';
import Calculator from './Calculator';

const CalculatorPage = () => {
  const [calcNumbers] = useState(numbers);
  const [calcOperators] = useState(operators);

  return (
    <div className="calcpage" data-testid="calc-1">
      <div className="calctitle" data-testid="calc-2">Lets do some math!</div>
      <Calculator operators={calcOperators} numbers={calcNumbers} />
    </div>
  );
};

export default CalculatorPage;
