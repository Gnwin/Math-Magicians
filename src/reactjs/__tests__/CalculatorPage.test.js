import {render, screen} from '@testing-library/react'
import renderer from 'react-test-renderer';
import '@testing-library/jest-dom'

import CalculatorPage from '../components/CalculatorPage';


describe('it tests the CalculatorPage Component', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(<CalculatorPage />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('checks if calcpageTitle title is a descendant of calcpageComponent', () => {
    render(<CalculatorPage />);
    const calcpageComponent = screen.getByTestId('calc-1');
    const calcpageTitle = screen.getByTestId('calc-2');
    expect(calcpageComponent).toContainElement(calcpageTitle);
  });
});