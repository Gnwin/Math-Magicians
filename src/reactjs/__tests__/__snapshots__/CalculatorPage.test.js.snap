// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it tests the CalculatorPage Component renders correctly 1`] = `
<div
  className="calcpage"
  data-testid="calc-1"
>
  <div
    className="calctitle"
    data-testid="calc-2"
  >
    Lets do some math!
  </div>
  <div
    className="calculator"
    data-testid="calc"
  >
    <div
      className="answer"
    >
      0
    </div>
    <div
      className="center"
    >
      AC
    </div>
    <div
      className="center"
    >
      +/-
    </div>
    <div
      className="center"
    >
      %
    </div>
    <div
      className="center orange"
    >
      รท
    </div>
    <div
      className="center"
    >
      7
    </div>
    <div
      className="center"
    >
      8
    </div>
    <div
      className="center"
    >
      9
    </div>
    <div
      className="center orange"
    >
      x
    </div>
    <div
      className="center"
    >
      4
    </div>
    <div
      className="center"
    >
      5
    </div>
    <div
      className="center"
    >
      6
    </div>
    <div
      className="center orange"
    >
      -
    </div>
    <div
      className="center"
    >
      1
    </div>
    <div
      className="center"
    >
      2
    </div>
    <div
      className="center"
    >
      3
    </div>
    <div
      className="center orange"
    >
      +
    </div>
    <div
      className="center zero"
    >
      0
    </div>
    <div
      className="center"
    >
      .
    </div>
    <div
      className="center orange"
    >
      =
    </div>
  </div>
</div>
`;
